fdbclient/BackupAgent.actor.h

```diff
@@ -883,7 +883,7 @@ class BackupConfig : public KeyBackedConfig {
 		}
 		TraceEvent t(SevWarn, "FileBackupError");
 		t.error(e).detail("BackupUID", uid).detail("Description", details).detail("TaskInstance", (uint64_t)taskInstance);
-		// These should not happen
+		// key_not_found could happen
 		if(e.code() == error_code_key_not_found)
```

Comment:
I added an assert here `ASSERT(e.code() != error_code_key_not_found)` and it failed in correctness.

Reply:
The original comment is describing an error scenario, which is probably something we want to know why. So I am leaning toward not changing the comment.

Reply:
It's used in the old backup and restore extensively. Basically, it uses "throw and error" to trigger a retry. 
The old comment is confusingly incorrect, indicating that the error should not happen.